packages/back-end/src/services/auth/OpenIdAuthConnection.ts

```diff
@@ -159,19 +161,22 @@ export class OpenIdAuthConnection implements AuthConnection {
         );
       }
 
-      const middleware = jwtExpress({
-        secret: jwks.expressJwtSecret({
-          cache: true,
-          cacheMaxEntries: 50,
-          rateLimit: true,
-          jwksRequestsPerMinute: 5,
-          jwksUri,
-        }),
-        audience: connection.clientId,
-        issuer,
-        algorithms,
-      });
-      middleware(req as Request, res, next);
+      if (!jwksClients[jwksUri]) {
```

Comment:
I think we'll want a compound key here to be safe that hashes together `jwksUri`, `connection.clientId`, `issuer`, and maybe also `algorithms`.  Some IdPs like Google have a single shared jwksUri for everyone and the clientId is the only way to differentiate between tenants.